### PR TITLE
AB#94091 Made price inputs in AHP not allow decimals

### PR DESCRIPTION
**Pull request by MichalSzrederPWC**
### 🛠 Changes being made

Made price inputs in AHP whole number only.
Removed PoundsPenceValueObject because it was no longer used.
Replaced all usages of PoundValueObject with TheRequiredIntValueObject and removed PoundValueObject.
Added unit tests for ExpectedPurchasePrice and PurchasePrice.
Changed validation error for invalid number range to more descriptive one.
Verified correctness of values not being decimals with https://dev.azure.com.mcas.ms/homesengland/Investments%20Programme/_workitems/edit/65922?McasTsid=26110&McasCtx=4 (explicitly states that fields should be whole pounds)
https://dev.azure.com.mcas.ms/homesengland/Investments%20Programme/_workitems/edit/65923?McasTsid=26110&McasCtx=4 (doesn't state whether public grant values should be decimals or whole pounds)

### 📸 Screenshots (optional)

image
image
image
image
image
image
image

### 🏎 Quality check

- [x] Have you performed local tests?
- [x] Are integration tests passing locally?
- [x] Have you added some unit tests?
- [ ] Have you added some integration tests?
- [ ] Have you checked WCAG (included screenshot)

### CODE REVIEW

**Review by Carq — APPROVED** [review]

File: src/HE.Investments.Common/Validators/WholeNumberRangeValidator.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+namespace HE.Investments.Common.Validators;
+
+public static class WholeNumberRangeValidator
+{
+    public static decimal ValidateDeffer(decimal number, string fieldName, string displayName, OperationResult operationResult)
+    {
+        number = Math.Round(number, 0);
+        if (number is < 0 or > 999999999)
+        {
+            operationResult.AddValidationError(fieldName, ValidationErrorMessage.MustBeWholeNumberBetween(displayName, 0, 999999999));
```
Comment (jjankowski87) [inline src/HE.Investments.Common/Validators/WholeNumberRangeValidator.cs:13]: Please differentiate too low and too high value with MustProvideTheHigherNumber and MustProvideTheLowerNumber error messages.

**Review by jjankowski87 — APPROVED** [review]
On some screens line on the left is too long, can you fix that as well?

**Review by dawid-pawulski — APPROVED** [review]

File: src/HE.Investments.Common/Validators/WholeNumberRangeValidator.cs (hunk 1 above)
Comment (MichalSzrederPWC) [reply src/HE.Investments.Common/Validators/WholeNumberRangeValidator.cs]: Replaced usages of PundsValueObject with TheRequiredIntValueObject everywhere.
TheRequiredIntValueObject uses MustProvideTheHigherNumber and MustProvideTheLowerNumber for range validation.

**Comment by MichalSzrederPWC** [comment]
> On some screens line on the left is too long, can you fix that as well?

image

**Review by jjankowski87 — APPROVED** [review]

**Pull request merged**